src/main/com/yetanalytics/datasim/io.clj

```diff
@@ -11,7 +11,8 @@
          (try
            (p/read-body-fn
             record
-            (json/parse-stream r (partial p/read-key-fn record)))
+            (doall ;; Force eager parsing of the entire file
+             (json/parse-stream r (partial p/read-key-fn record))))
```

Comment:
Might want to mention that it's specifically Cheshire that does not like lazy seqs.

Reply:
I think it is sufficiently explicit, this is actually stated behavior for `cheshire.core/parse-stream`